packages/client/src/Views/GameSetup.jsx

```diff
@@ -1,146 +1,134 @@
 
-import classNames from "classnames";
-import TextInput from "../Components/Inputs/TextInput";
-import {addNotification} from "../ActionsAndReducers/Notification/Notification_ActionCreators";
-import {ADMIN_ROUTE} from "../consts";
+import _ from 'lodash'
+import checkUnique from '../Helpers/checkUnique'
 
-class GameSetup extends Component {
+import classNames from 'classnames'
+import TextInput from '../Components/Inputs/TextInput'
+import { addNotification } from '../ActionsAndReducers/Notification/Notification_ActionCreators'
+import { ADMIN_ROUTE } from '../consts'
 
-  constructor(props) {
-    super(props);
+const GameSetup = ({ wargame, dispatch }) => {
+  const [newWargameName, setNewWargameName] = useState(null)
 
-    this.state = {
-      newWargameName: null,
-    };
-  }
-
-  setCurrentTab = (tab) => {
-    this.props.dispatch(setCurrentTab(tab));
-  };
-
-  createIndicators() {
-    const that = this;
-
-    return Object.entries(this.props.wargame.data).map(function (entry) {
-      let active = that.props.wargame.currentTab === entry[0];
-      let completed = entry[1].complete;
-      return (<div key={entry[0]} className={classNames("progress-indicator", {"progress-indicator--active": active, "progress-indicator--complete": completed})}></div>);
-    });
-  }
-
-  updateWargameTitle = (name) => {
-    this.setState({
-      newWargameName: name,
-    });
-  };
+  const saveWargame = () => {
+    let wargameNames = wargame.wargameList.map(game => game.title)
+    wargameNames = _.pull(wargameNames, wargame.wargameTitle)
 
+    if (!checkUnique(newWargameName, wargameNames)) {
+      dispatch(addNotification('Wargame name is not unique.', 'warning'))
+      return
+    }
 
-  saveWargame = () => {
+    if (!checkWargameNameSaveable()) return
 
-    let wargameNames = this.props.wargame.wargameList.map((game) => game.title);
-        wargameNames = _.pull(wargameNames, this.props.wargame.wargameTitle);
+    if (typeof newWargameName === 'string' && newWargameName.length > 0) {
+      dispatch(saveWargameTitle(wargame.currentWargame, newWargameName))
+    }
 
-    if (!checkUnique(this.state.newWargameName, wargameNames)) {
-      this.props.dispatch(addNotification("Wargame name is not unique.", "warning"));
-      return;
+    if (newWargameName === null || newWargameName.length === 0) {
+      // TODO: We should replace this with a notification component
+      window.alert('no channel name')
     }
+  }
 
-    if (!this.checkWargameNameSaveable()) return;
+  const currentTabSetter = tab => dispatch(setCurrentTab(tab))
 
-    if (typeof this.state.newWargameName === 'string' && this.state.newWargameName.length > 0) {
-      this.props.dispatch(saveWargameTitle(this.props.wargame.currentWargame, this.state.newWargameName));
+  // Loop through wargame data and build up an array of tab objects
+  const allTabs = () => Object.entries(wargame.data).map(tab => {
+    console.log('running')
```

Comment:
@foxleigh81  - can the above debug line be removed?

Reply:
Whoops. I'll fix that in a sec.